File: LibraryViewController.m
//
//  LibraryViewController.m
//  Manhattan College App
//
//  Created by Nick Monaco on 2/3/14.
//  Copyright (c) 2014 Monaco Software Productions. All rights reserved.
//

#import "LibraryViewController.h"

@interface LibraryViewController () <UIWebViewDelegate, UIScrollViewDelegate>

@end

@implementation LibraryViewController
@synthesize LibWebView;

- (id)initWithNibName:(NSString *)nibNameOrNil bundle:(NSBundle *)nibBundleOrNil
{
    self = [super initWithNibName:nibNameOrNil bundle:nibBundleOrNil];
    if (self) {
        // Custom initialization
    }
    return self;
}

- (void)viewDidLoad
{
    [super viewDidLoad];
    NSString *fullURL = @"http://lib.manhattan.edu/";
    NSURL *url = [NSURL URLWithString:fullURL];
    NSURLRequest *requestObj = [NSURLRequest requestWithURL:url];
    [LibWebView loadRequest:requestObj];
	// Do any additional setup after loading the view.
    
    self.LibWebView.delegate = self;
    
    LibWebView.scrollView.delegate = self;
}

- (void)didReceiveMemoryWarning
{
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}

-(IBAction)openActionSheet:(id)sender;
{
    UIAlertController *actionSheet = [UIAlertController alertControllerWithTitle:nil message:nil preferredStyle:UIAlertControllerStyleActionSheet];
    
    [actionSheet addAction:[UIAlertAction actionWithTitle:@"Cancel" style:UIAlertActionStyleCancel handler:^(UIAlertAction *action) {
        
        // Cancel button tappped.
        //        [self dismissViewControllerAnimated:YES completion:^{
        //        }];
    }]];
    
    [actionSheet addAction:[UIAlertAction actionWithTitle:@"Open In Safari" style:UIAlertActionStyleDefault handler:^(UIAlertAction *action) {
        
        // OK button tapped.
        NSString *currentURL = LibWebView.request.URL.absoluteString;
        
        [[UIApplication sharedApplication] openURL:[NSURL URLWithString:currentURL]];
        
        [self dismissViewControllerAnimated:YES completion:^{
        }];
    }]];
    // Present action sheet.
    [self presentViewController:actionSheet animated:YES completion:nil];
}

- (void)updateButtons
{
    self.forward.enabled = self.LibWebView.canGoForward;
    self.back.enabled = self.LibWebView.canGoBack;
}

- (void)webViewDidStartLoad:(UIWebView *)webView
{
    [UIApplication sharedApplication].networkActivityIndicatorVisible = YES;
    [self updateButtons];
}
- (void)webViewDidFinishLoad:(UIWebView *)webView
{
    [UIApplication sharedApplication].networkActivityIndicatorVisible = NO;
    [self updateButtons];
}
- (void)webView:(UIWebView *)webView didFailLoadWithError:(NSError *)error
{
    [UIApplication sharedApplication].networkActivityIndicatorVisible = NO;
    [self updateButtons];
}


@end

File: Manhattan College App/LibraryViewController.swift
//
//  LibraryViewController.swift
//  MC
//
//  Created by Nick Monaco on 9/22/16.
//  Copyright © 2016 Monaco Software Productions. All rights reserved.
//

import Foundation
import UIKit

class LibraryViewController: UIViewController, UIWebViewDelegate, UIActionSheetDelegate, UIScrollViewDelegate {
    @IBOutlet var libWebView: UIWebView!
    @IBOutlet weak var back: UIBarButtonItem!
    @IBOutlet weak var forward: UIBarButtonItem!
    
    override init(nibName nibNameOrNil: String?, bundle nibBundleOrNil: Bundle?) {
        super.init(nibName: nibNameOrNil, bundle: nibBundleOrNil)
        
        // Custom initialization
        
    }
    
    required init?(coder aDecoder: NSCoder) {
        super.init(coder: aDecoder)
    }
    
    override func viewDidLoad() {
        super.viewDidLoad()
        let fullURL = "http://inside.manhattan.edu/"
        let url = NSURL(string: fullURL)!
        let requestObj = URLRequest(url: url as URL)
        libWebView.loadRequest(requestObj)
        // Do any additional setup after loading the view.
        //self.libWebView.delegate! = self
        //libWebView.scrollView.delegate = self
    }
    
    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        // Dispose of any resources that can be recreated.
    }
    
    @IBAction func openActionSheet(_ sender: Any) {
        let actionSheet = UIAlertController(title: nil, message: nil, preferredStyle: .actionSheet)
        actionSheet.addAction(UIAlertAction(title: "Cancel", style: .cancel, handler: {(action: UIAlertAction) -> Void in
            // Cancel button tappped.
            //        [self dismissViewControllerAnimated:YES completion:^{
            //        }];
        }))
        actionSheet.addAction(UIAlertAction(title: "Open In Safari", style: .default, handler: {(action: UIAlertAction) -> Void in
            // OK button tapped.
            let currentURL = self.libWebView.request!.url?.absoluteString
            UIApplication.shared.openURL(NSURL(string: currentURL!)! as URL)
            self.dismiss(animated: true, completion: {() -> Void in
            })
        }))
        // Present action sheet.
        self.present(actionSheet, animated: true, completion: { _ in })
    }
    
    func updateButtons() {
        self.forward.isEnabled = self.libWebView.canGoForward
        self.back.isEnabled = self.libWebView.canGoBack
    }
    
    func webViewDidStartLoad(_ webView: UIWebView) {
        UIApplication.shared.isNetworkActivityIndicatorVisible = true
        self.updateButtons()
    }
    
    func webViewDidFinishLoad(_ webView: UIWebView) {
        UIApplication.shared.isNetworkActivityIndicatorVisible = false
        self.updateButtons()
    }
    
    func webView(_ webView: UIWebView, didFailLoadWithError error: Error) {
        UIApplication.shared.isNetworkActivityIndicatorVisible = false
        self.updateButtons()
    }
}
